lib/util/node-event-generator.js

```diff
@@ -46,6 +67,7 @@ NodeEventGenerator.prototype = {
      * @returns {void}
      */
     leaveNode: function leaveNode(node) {
+        this.currentAncestry.shift();
```

Comment:
It looks like this implementation will not support `FunctionDeclaration>Identifier:exit` the way it is right now. But that can be added later. Just something to keep in mind.